### PR TITLE
Set up Next.js hello-world app with dev tools

### LINKED ISSUE #1: Set up Next.js hello-world app with dev tools
# Overview

We've done some initial brainstorming, and now we're ready to kick off this project! The goal of this issue's work is to enable others to start contributing. Next.js needs to be set up, as well as some development tools like Prettier and ESLint.

# Acceptance criteria

- [ ] The app is a Next.js project that can be run locally
- [ ] CSS infrastructure is set up (fonts, colors, utilities, resets, etc.)
- [ ] Development tools are installed and active:
  - [ ] Prettier (formatting)
  - [ ] ESLint (JS & TS linting)
  - [ ] Stylelint (CSS linting)
  - [ ] lint-staged (run linters against staged git files)
  - [ ] Husky (commit hooks)
- [ ] Format & lint checks are run in a pre commit hook
- [ ] Format & lint checks are run on PRs

### PR DESCRIPTION
**Pull request by ashleemboyer**
## Relevant issues

- Closes #1 

## What

- Adds initial Contributing and Development docs to the main `README.md`
- Sets up [Next.js](https://nextjs.org/) with the [Pages Router](https://nextjs.org/docs/pages) to start
  - `src/css/` sets up variables, colors, fonts, utilities, and initial styles
  - `src/pages/_app.tsx` imports the global stylesheet (see [Next.js docs on global styling](https://nextjs.org/docs/pages/building-your-application/styling/css-modules#global-styles))
  - `src/pages/index.tsx` sets up the homepage with basic info about the group
- Adds development tools:
  - [EditorConfig](https://editorconfig.org/)
  - [ESLint](https://eslint.org/)
  - [Prettier](https://prettier.io/)
  - [Stylelint](https://stylelint.io/)
  - [Husky](https://github.com/typicode/husky#readme)
  - [lint-staged](https://github.com/lint-staged/lint-staged)
  - [PostCSS](https://postcss.org/)
  - [TypeScript](https://www.typescriptlang.org/)
- Adds list of VSCode extensions highly recommended for working with the project
- Adds `package.json` scripts for running "check" and "fix" commands for prettier, eslint, and stylelint
- Adds a pre-commit hook that runs prettier, eslint, and stylelint on files staged in a commit (see `.lintstagedrc.js`)
- Adds CI (continuous integration) GitHub workflow for checking code format and style in pull requests

## Why

- Since the Next.js [App Router](https://nextjs.org/docs/app) paradigm is still on the newer side, we're going to start with Pages Router so folks without Next.js experience can build a good initial understanding. After we get our first few pages launched, we'll look at how we can use App Router on additional pages.
- EditorConfig, ESLint, Prettier, and Stylelint are common development tools for maintaining consistent code format and styling. When using the recommended VSCode extensions (see `.vscode/extensions.json`), any issues they find _should_ be auto-fixed on save. As a back up to that, Husky's pre-commit hook and lint-staged will check for and fix issues found on files staged for commit when `git commit` is run.
- PostCSS is manually configured so we can use the `postcss-nesting`, `postcss-import-ext-glob`, and `postcss-import` plugins. Next.js [disables the default behavior](https://nextjs.org/docs/pages/building-your-application/configuring/post-css#customizing-plugins) when a custom PostCSS file is in the project, so we have to include the default configuration in our file as well.

## Testing

- [ ] Make sure the `README.md` makes sense & is accurate
- [ ] Check out this branch locally (follow the fork & clone steps in `README.md` if needed)
- [ ] Run `npm install`
- [ ] Dependencies should successfully install
- [ ] Run `npm start`
- [ ] Go to http://localhost:3000/ in your browser
- [ ] The app should look something like this (depending on your window size):

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,46 @@
...
+
 The website for React.Indy - an Indianapolis-based group for anyone interested in React, React Native, React frameworks (like Next.js), JavaScript, TypeScript, or front-end engineering.
+
+## 1. Contributing
+
+The contributing process documentation is still in-progress. The numbered list below has the basics for the meantime. If you're unsure or stuck about a particular step, first see the ["Related resources"](#1b-related-resources) section directly after the list. If you still need help or reassurance, just ask! :)
+
+### 1a. The general flow
+
+After you've created a fork of this repository under your GitHub account, and cloned your fork to your machine, the contribution flow looks like:
```
Comment (ejmudrak) [inline README.md:11]: Contributing via forking the repo sounds good to me! That'd certainly be less to manage on your end if anyone would like to open a PR.

File: src/css/design-tokens/colors.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/*
+* Color names come from the Coolors palette tool:
+* https://coolors.co/222222-3b3b3b-023f50-6adcfb-ffffff
```
Comment (ejmudrak) [inline src/css/design-tokens/colors.css:3]: Thanks for adding helpful links in comments, like the color palette here

File: src/pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+const CODE_OF_CONDUCT_LINK =
+	"https://docs.google.com/document/d/e/2PACX-1vQf_7z4Ok-zyUY9vej4ZDafitByM6TOuZ8ju2rcEdEuKtuTYYrtDNjjw0EZRmt9pVQ2k7PqghYf16-F/pub";
+
+const HomePage = () => {
+	return (
+		<div className={styles.root}>
+			<header className={styles.header}>
+				{/* eslint-disable-next-line @next/next/no-img-element */}
+				<img
```
Comment (ejmudrak) [inline src/pages/index.tsx:12]: Why not use Next's `Image` component here?

**Review by ejmudrak — COMMENTED** [review]
This is a solid foundation! Thanks for your initiative and care for documentation here.

File: src/pages/index.tsx (hunk 3 above)
Comment (ashleemboyer) [reply src/pages/index.tsx]: I didn't want to have to define the `width` and `height` properties, but now remember the `next/image` way around that is by [statically importing the image file](https://nextjs.org/docs/app/building-your-application/optimizing/images#local-images) and passing it in through `src`!

Updated in https://github.com/React-Indy/website/pull/2/commits/f958b9b8ce96d24da08722d1e5c3bf0e70507fc9. Great callout 💯

**Review by ejmudrak — APPROVED** [review]
Approved!

**Comment by ashleemboyer** [comment]
Woot! 🎉

Thanks for the review @ejmudrak, and for helping me test out the repo member permissions!

**Pull request merged**